setup.py

```diff
@@ -1,31 +1,103 @@
-import setuptools
+"""
+jupyter-server-proxy setup
+"""
+import json
 from glob import glob
+from pathlib import Path
+
+import setuptools
+from jupyter_packaging import (
+    combine_commands,
+    create_cmdclass,
+    ensure_targets,
+    install_npm,
+    skip_if_exists,
+)
+
+HERE = Path(__file__).parent.resolve()
+
+# The name of the project
+name = "jupyter_server_proxy"
+
+lab_path = HERE / name / "labextension"
+
+# Representative files that should exist after a successful build
+jstargets = [
+    str(lab_path / "package.json"),
+]
+
+package_data_spec = {
+    name: ["*"],
+}
+
+labext_name = "jupyter_server_proxy"
+
+data_files_spec = [
+    ("share/jupyter/labextensions/%s" % labext_name, str(lab_path), "**"),
+    ("share/jupyter/labextensions/%s" % labext_name, str(HERE), "install.json"),
+    # {%- if cookiecutter.has_server_extension == "y" -%}
+    # ("etc/jupyter/jupyter_server_config.d",
+    #  "jupyter-config", "{{ cookiecutter.python_name }}.json"),
+    # {% endif %}
+]
 
-with open("README.md") as f:
-    readme = f.read()
-
-setuptools.setup(
-    name="jupyter-server-proxy",
-    version='1.6.0',
-    url="https://github.com/jupyterhub/jupyter-server-proxy",
-    author="Ryan Lovett & Yuvi Panda",
-    author_email="rylo@berkeley.edu",
-    license="BSD 3-Clause",
-    description="Jupyter server extension to supervise and proxy web services",
-    long_description=readme,
+cmdclass = create_cmdclass(
+    "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE / "jupyterlab-server-proxy", build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "jupyterlab-server-proxy" / "package.json").read_bytes())
+
+setup_args = dict(
+    name=name,
+    version=pkg_json["version"],
+    url=pkg_json["homepage"],
+    author=pkg_json["author"],
+    description=pkg_json["description"],
+    license=pkg_json["license"],
```

Comment:
The metadata between `package.json` and `setup.py` differs. Is this intended? If yes, I would hard code the original values here. Otherwise, I would suggest this approach, as it avoids redundancy.

Reply:
I have adjusted it to reflect the original values from `setup.py`.